src/main/webapp/intrigue-api/context.js

```diff
@@ -1,18 +1,22 @@
 import createRpcClient from './rpc'
 import fetch from './fetch'
+import { AuthenticationError } from 'apollo-server-errors'
 
 const context = args => {
   const { req } = args
 
   const { authorization = '' } = req.headers
-  const universalFetch = (url, opts = {}) => {
-    return fetch(url, {
+  const universalFetch = async (url, opts = {}) => {
+    const res = await fetch(url, {
       ...opts,
       headers: {
         ...opts.headers,
         authorization,
       },
     })
+    if (res.status === 401) {
```

Comment:
Going to add more functionality for different kinds of errors. Authentication is just the first one.